### PR TITLE
Remove Avenir Next Demibold web font, replace with Medium + CSS effects

### PR DESCRIPTION
**Pull request by cfarm**
Removes Avenir Next Demibold web font to improve performance for end users. Demibold is replaced with Avenir Next Medium + CSS effects to maintain the existing type hierarchy. For more background see https://GHE/CFGOV/platform/issues/2347#issuecomment-144964 

## Additions

-

## Removals

-

## Changes

Text elements previously using Demibold are mainly 2 types: h5-style headings, and inline bold text. Both of these were updated as part of this PR.

### H5 and elements that inherit its styling:

h5 and h6 now use Avenir Next Medium plus a CSS `text-shadow` effect to mimic the thicker weight of Demibold as compared to Medium. Additional elements inherit this CSS:

- `th` (Table heading) elements on tables (cf-core)
- Responsive stacked table headings (cf-tables)
- Slug header (cf-typography)
- Pull quote author - text-shadow color is set to match the gray text color instead of default black of heading-5 elements (cf-typography)
- Date, with text-shadow updated to gray (cf-typography)

### Inline bold:

Inline bolded text - e.g., text in paragraphs **wrapped in `<b>` or `<strong>` tags for emphasis** - was previously styled using Demibold, under the `.u-webfont-demi` class. To replace the web font but maintain the level of contrast needed between **bold text** and regular text, the Avenir Next Medium font is used, with `font-weight: bold` added to the class. The class name `u-webfont-demi` is still used to maintain consistency with our print and visual design type hierarchy. We just fake the demibold without actually making the user load that extra font file. 😎 

## Testing

Use [testing instructions](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-components-locally) to run this locally with CF docs and confirm the above elements match screenshots for the following component docs:

- http://localhost:3000/components/cf-typography/
- http://localhost:3000/components/cf-tables/
- http://localhost:3000/components/cf-core/

## Screenshots

To make this PR easier to review I'm posting screenshots in the comments.

## Notes

-

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### Browsers

- [ ] Chrome
- [ ] Safari
- [ ] FF
- [ ] IE10
- [ ] IE9
- [ ] IE8
- [ ] Opera
- [ ] iOS
- [ ] Android
- [ ] Blackberry Bold

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: src/cf-typography/src/cf-typography.less [hunk 1]
```diff
@@ -11,7 +11,8 @@
 @webfont-regular: 'AvenirNextLTW01-Regular';
 @webfont-italic: @webfont-regular;
 @webfont-medium: 'AvenirNextLTW01-Medium';
-@webfont-demi: 'AvenirNextLTW01-Demi';
+@webfont-demi: @webfont-regular;
+// @webfont-demi: 'AvenirNextLTW01-Demi';
```
Comment (anselmbradford) [inline src/cf-typography/src/cf-typography.less]: Can this be deleted?

Comment (cfarm) [reply src/cf-typography/src/cf-typography.less]: Yep, that's why this is still a WIP :) I'll ping you when it's ready for review!

**Comment by cfarm** [comment]
# Demibold replacement for headings

## Pull quote author
demibold replacement default pull quote author

## Date
demibold replacement date

## Slug header
demibold replacement slug header

## Table header
demibold replacement table

## H5 and H6 headings
demibold replacement h5 h6

**Comment by cfarm** [comment]
# Fallback font (Arial)
Here is how the updated styles appear when Avenir Next is not loaded:

## h5 and h6
demibold replacement arial h5 h6

## Table header
demibold replacement arial table

File: src/cf-typography/src/atoms/date.less [hunk 2]
```diff
@@ -2,5 +2,6 @@
     .heading-5();
 
     color: @date;
+    text-shadow: 0px 0px @date;
```
Comment (anselmbradford) [inline src/cf-typography/src/atoms/date.less]: Won't have IE8/9 support, which is fine with me, but just documenting.

File: src/cf-core/src/cf-base.less [hunk 3]
```diff
@@ -103,12 +102,14 @@ body {
...
     @font-size: @fs;
-    .u-webfont-demi();
+    .u-webfont-medium();
 
     margin-bottom: unit( 15px / @font-size, em );
     font-size: unit( @font-size / @base-font-size-px, em );
+    font-weight: bold;
     letter-spacing: 1px;
     line-height: 1.25;
+    text-shadow: 0px 0px @text;
```
Comment (jimmynotjim) [inline src/cf-core/src/cf-base.less]: Can we move the color to a param so users don't have to know that they need to over-write this when they change the text color?

File: src/cf-typography/src/cf-typography.less [hunk 4]
```diff
@@ -11,7 +11,7 @@
 @webfont-regular: 'AvenirNextLTW01-Regular';
 @webfont-italic: @webfont-regular;
 @webfont-medium: 'AvenirNextLTW01-Medium';
-@webfont-demi: 'AvenirNextLTW01-Demi';
+@webfont-demi: @webfont-regular;
```
Comment (jimmynotjim) [inline src/cf-typography/src/cf-typography.less]: Should this be `@webfont-medium`?

File: src/cf-typography/usage.md [hunk 5]
```diff
@@ -47,7 +47,7 @@ project by duplicating the variable `@key: value`.
 @webfont-regular: 'AvenirNextLTW01-Regular';
 @webfont-italic: @webfont-regular;
 @webfont-medium: 'AvenirNextLTW01-Medium';
-@webfont-demi: 'AvenirNextLTW01-Demi';
+@webfont-demi: @webfont-regular;
```
Comment (jimmynotjim) [inline src/cf-typography/usage.md]: Same thing here about `@webfont-medium`

File: src/cf-core/src/cf-base.less (hunk 3 above)
Comment (cfarm) [reply src/cf-core/src/cf-base.less]: I thought about this, I'll give it a try and set the default value to @text. Thanks for the nudge!

**Comment by cfarm** [comment]
OK, it looks like this PR is heavily impacted by the changes from https://github.com/cfpb/capital-framework/pull/732/files. I have some conflicts I don't know how to resolve. @Scotchester can you take a look or post some guidance on how I should incorporate your changes?

**Comment by cfarm** [comment]
@Scotchester can you hold off on updating cfgov-refresh with the changes from #732 until I can fix this conflict and release my own changes? It will help us avoid duplicate work (having to update all the downstream classes that used `.u-webfont-demi` twice).

File: src/cf-core/src/cf-base.less [hunk 6]
```diff
@@ -116,14 +115,15 @@ textarea {
     line-height: 1.25;
 }
 
-.heading-5( @fs: @size-v )  {
+.heading-5( @fs: @size-v, @text-shadow: @text )  {
```
Comment (jimmynotjim) [inline src/cf-core/src/cf-base.less]: Perfecto, thank you 👌

**Pull request merged**